### PR TITLE
Fix issue with some opcodes crashing on linkage

### PR DESCRIPTION
**Pull request by 2xic**
Fixes
- Reported issue by user #345 
- Links to opcodes not working if a fork was selected
- Forks not loading correctly because of race condition where forks were loaded after `onfork` was called. Fixed by adding a new field to the context.

Some of the underlying logic related to this should be re-worked as it's called from multiple places, but that is not in scope for this PR. 

## Test plan
1. Opening shared opcodes should not crash app (example from issue) 
      - (preview build works) https://evm-codes-git-hotfix-fix-link-crash-smlxl.vercel.app/?fork=cancun#f3 
      - (prod fails) https://evm.codes/?fork=cancun#f3
      - Screenshot 2024-09-16 at 19 10 40

2. Fork selector should still work
      - https://evm-codes-git-hotfix-fix-link-crash-smlxl.vercel.app/?fork=grayGlacier should not show `PUSH0`
      - https://evm-codes-git-hotfix-fix-link-crash-smlxl.vercel.app/?fork=shanghai should show `PUSH0` as it was added here 
      - Screenshot 2024-09-16 at 13 19 50

3. EOF opcode linking should also work
      - https://evm-codes-git-hotfix-fix-link-crash-smlxl.vercel.app/?fork=EOF#ec
      - https://evm-codes-git-hotfix-fix-link-crash-smlxl.vercel.app/?fork=EOF#f9
      - https://evm-codes-git-hotfix-fix-link-crash-smlxl.vercel.app/?fork=EOF#f3

4. Precompiles linking should also work
     - https://evm-codes-git-hotfix-fix-link-crash-smlxl.vercel.app/precompiled?fork=EOF#0x0a
         - Should show 0x0a as it was added later on 
     - https://evm-codes-git-hotfix-fix-link-crash-smlxl.vercel.app/precompiled?fork=cancun#0x0a
         - Should show 0x0a as it was added later on 
     - https://evm-codes-git-hotfix-fix-link-crash-smlxl.vercel.app/precompiled?fork=london#0x08
         - Should not show 0x0a as it was added later on 
         - Screenshot 2024-09-16 at 16 10 57

### CODE REVIEW

**Comment by 2xic** [comment]
* **#347** <a href="https://app.graphite.dev/github/pr/smlxl/evm.codes/347?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @2xic and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

**Review by 2xic — COMMENTED** [review]
some fyi comments

File: components/Editor/index.tsx [hunk 1]
```diff
@@ -240,7 +241,7 @@ const Editor = ({ readOnly = false }: Props) => {
       onForkChange(query.fork as string)
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [settingsLoaded && router.isReady])
+  }, [settingsLoaded && router.isReady && isForksLoaded])
```
Comment (mcmxcdev) [inline components/Editor/index.tsx:244]: This is not how one specifies hook dependencies, see: https://react.dev/reference/react/useEffect#useeffect

Its comma-separated values that react reads out, not a boolean condition like here.

You can verify this easily by letting intellisense (the light bulb) in e.g. VSCode autofix missing deps here.

File: components/Reference/index.tsx [hunk 2]
```diff
@@ -112,12 +113,11 @@ const ReferenceTable = ({
   // Change selectedFork according to query param
   useEffect(() => {
     const query = router.query
-
     if ('fork' in query) {
       onForkChange(query.fork as string)
     }
     // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [router.isReady])
+  }, [router.isReady && isForksLoaded])
```
Comment (mcmxcdev) [inline components/Reference/index.tsx:120]: Same here

File: context/ethereumContext.tsx [hunk 3]
```diff
@@ -79,6 +79,7 @@ type ContextProps = {
   isExecuting: boolean
   executionState: IExecutionState
   vmError: string | undefined
+  isForksLoaded: boolean
```
Comment (mcmxcdev) [inline context/ethereumContext.tsx:82]: Probably plural wording?

```suggestion
  areForksLoaded: boolean
```

File: package.json [hunk 4]
```diff
@@ -105,14 +106,14 @@
     "typescript": "^5.0.4",
     "webpack": "^5.90.3"
   },
-  "packageManager": "pnpm@8.15.6",
+  "packageManager": "pnpm@9.4.0",
```
Comment (mcmxcdev) [inline package.json:109]: Seems like a change that should be done in a separate PR?

Also, latest pnpm version right now is 9.10.0

If you needed these changes to be able to work locally, consider simply undoing them again before merging the PR.

**Review by ilamanov — DISMISSED** [review]
Code looks good. And tested according to the test plan - all good

**Review by mcmxcdev — DISMISSED** [review]

**Review by mcmxcdev — APPROVED** [review]

**Comment by 2xic** [comment]
### Merge activity

* **Sep 16, 7:12 PM GMT+2**: @2xic started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/smlxl/evm.codes/347).

File: components/Reference/index.tsx (hunk 2 above)
Comment (charisra) [reply components/Reference/index.tsx]: @2xic unless you find it bothersome, may you please not mark convos as resolved and save us a few clicks 😸 . You can acknowledge by a 👍 and if you need to 👀 first until you acknowledge, you can 👀 and then remove and 👍 .

Comment (charisra) [reply components/Reference/index.tsx]: I remember we discussed it in FE sync, not a requirement, seemed to be the consensus amongst the team back then.

Comment (2xic) [reply components/Reference/index.tsx]: Yup, I know, but what I'm saying is that this repo (evm.codes) required us to resolve all conversations before we could merge (like some other repos we have). So even I if I wanted to follow the rules, I couldn't 😅

Anyways - I have updated the settings now so that the evm.codes repo settings matches our other FE monorepo.

Comment (charisra) [reply components/Reference/index.tsx]: 🤦 my bad, didn't notice the repo...
